src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,121 @@
+
+#### NUpbr
+
+NUpbr is a Physically Based Rendering tool created in Blender. It creates semi-synthetic images with corresponding segmentation images for training.
+
+Find out how to get and use NUpbr on the [NUpbr NUbook page](/system/tools/nupbr) and the [NUpbr GitHub repository](https://github.com/NUbots/NUpbr).
+
+#### Setting Up The Data
+
+The [Visual Mesh repository](https://github.com/Fastcode/VisualMesh) contains information on how to set up the datasets for training. It contains [a script](https://github.com/Fastcode/VisualMesh/blob/master/training/build_dataset.py) that takes data and converts it into the tfrecord format that the Visual Mesh expects. The script requires a folder of raw images, segmentation masks, and metadata. NUpbr will provide all these as output, and such data can be found on the NAS.
+
+### The Visual Mesh
+
+#### Training and Testing
+
+Go to the [NUbook Visual Mesh Getting Started guide](/guides/tools/visualmesh) to find out how to train and test a network, with an example dummy dataset.
+
+#### Exporting Configuration
+
+The resulting networking should be exported to a [yaml](https://yaml.org/) file and added to the NUbots codebase.
+
+Example yaml files can be found [in the Visual Mesh repository](https://github.com/Fastcode/VisualMesh/blob/master/example/model.yaml) and [in the NUbots repository](https://github.com/NUbots/NUbots/blob/master/module/vision/VisualMesh/data/config/VisualMesh.yaml). Assuming such a file is in the output directory, run
+
+```bash
+./mesh.py export [output_dir]
+```
+
+to export the weights to the file.
+
+Replace the NUbots repository `VisualMesh.yaml` configuration with this configuration file. Ensure you create a new branch so you can make a PR and get any changes merged in. View the [Git Guide](/guides/general/git) for information on using Git.
+
+### Camera Calibration
+
+The vision system cannot work optimally if the cameras are not calibrated correctly. [The input page](/system/subsystems/input) describes the camera parameters that can be calibrated.
+
+An automatic camera calibration tool is available in the NUbots repository. See the [camera calibration guide](guides/main/camera-calibration#page-content) to find out how to use this tool.
+
+### Testing
+
+After updating the Visual Mesh in the NUbots repository, it should be tested before merging.
+
+1. Build the code, ensuring `ROLE_visualmesh` is set to `ON` in `./b configure -i`. Go to the [Getting Started page](/guides/main/getting-started) to find out more on how to build the code and install to a robot. Ensure the new configuration file has been installed. Check out the [Build System page](/system/foundations/build-system#install) to find out more about options when installing onto the robot.
+
+2. When your new Visual Mesh is installed onto the robot, connect to the robot using
+
+   ```bash
+   ssh nubots@<address>
```

Comment:
Please either clarify this more or link to the nubook page with more details

Reply:
I guess this is similar to the nano thing as well. I can link to [getting started](/guides/main/getting-started#running-on-a-real-robot) or just assume they'll search if they don't understand something, to reduce clutter. I think it might just be good to link to getting started at the beginning of the testing section.

Reply:
I agree with Ysobel. Linking to the getting started guide would be the best option as these docs should be specific to maintenance for vision and don't need to be stand-alone as the idea of search and 'generic' tutorials on NUbook is that info doesn't need to get repeated.